src/utils/updateFavouriteBooks.ts

```diff
@@ -0,0 +1,23 @@
+import type { Book } from "../types";
+
+export default function updateFavouriteBooks(book: Book) {
+  const favouritesFromStorage = localStorage.getItem("favourites");
+  let favourites: Book[];
+  if (favouritesFromStorage !== null) {
+    favourites = JSON.parse(favouritesFromStorage) as Book[];
+  } else {
+    favourites = [];
+  }
+  if (
+    favourites.filter((favourite: Book) => favourite.id === book.id).length > 0
```

Comment:
nie musisz tutaj dodawać adnotacji typów, typescript sam to ogarnie
```suggestion
    favourites.filter((favourite) => favourite.id === book.id).length > 0
```

oraz chyba lepiej tu użyć find lub indexOf

```suggestion
    favourites.find((favourite) => favourite.id === book.id) !== undefined
```